web/src/main/java/org/springframework/security/web/context/HttpSessionSecurityContextRepository.java

```diff
@@ -327,6 +330,9 @@ public AsyncContext startAsync(ServletRequest servletRequest, ServletResponse se
 		@Override
 		protected void saveContext(SecurityContext context) {
 			final Authentication authentication = context.getAuthentication();
+			if (isTransientAuthentication(authentication)) {
```

Comment:
By moving the call to `isTransientAuthentication` out and to the beginning of `saveContext`, it seems like `createNewSessionIfAllowed`'s implementation and contract can be simplified by removing its call to `isTransientAuthentication`.

Should `createNewSessionIfAllowed` continue to call `isTransientAuthentication`?

Reply:
Thanks this is a good point. I've removed the unnecessary check in `createNewSessionIfAllowed`. Can you please verify and merge if things look good now?

Reply:
It looks like the `Authentication` method parameter in `createNewSessionIfAllowed` is no longer necessary either. Can that be removed as well?